Gems/Atom/Tools/AtomToolsFramework/Code/Source/AssetSelection/AssetSelectionComboBox.cpp

```diff
@@ -78,34 +74,108 @@ namespace AtomToolsFramework
         setCurrentIndex(findData(QVariant(QString(assetId.ToString<AZStd::string>().c_str()))));
     }
 
+    void AssetSelectionComboBox::SetThumbnailsEnabled(bool enabled)
+    {
+        if (m_thumbnailsEnabled != enabled)
+        {
+            m_thumbnailKeys.clear();
+            m_thumbnailsEnabled = enabled;
+            for (int index = 0; index < count(); ++index)
+            {
+                setItemIcon(index, QIcon());
+                RegisterThumbnail(AZ::Data::AssetId::CreateString(itemData(index).toString().toUtf8().constData()));
+            }
+        }
+    }
+
+    void AssetSelectionComboBox::SetThumbnailDelayMs(AZ::u32 delay)
+    {
+        m_thumbnailDelayMs = delay;
+    }
+
     void AssetSelectionComboBox::OnCatalogAssetAdded(const AZ::Data::AssetId& assetId)
     {
         AZ::Data::AssetCatalogRequestBus::Broadcast(
             [this, assetId](AZ::Data::AssetCatalogRequests* assetCatalogRequests)
             {
-                AZ::Data::AssetInfo assetInfo = assetCatalogRequests->GetAssetInfoById(assetId);
-                if (m_filterCallback && m_filterCallback(assetInfo))
-                {
-                    addItem(
-                        GetDisplayNameFromPath(AZ::RPI::AssetUtils::GetSourcePathByAssetId(assetInfo.m_assetId).c_str()),
-                        QVariant(QString(assetInfo.m_assetId.ToString<AZStd::string>().c_str())));
-                }
+                AddAsset(assetCatalogRequests->GetAssetInfoById(assetId));
                 model()->sort(0, Qt::AscendingOrder);
             });
     }
 
-    void AssetSelectionComboBox::OnCatalogAssetRemoved(
-        [[maybe_unused]] const AZ::Data::AssetId& assetId, const AZ::Data::AssetInfo& assetInfo)
+    void AssetSelectionComboBox::OnCatalogAssetRemoved(const AZ::Data::AssetId& assetId, const AZ::Data::AssetInfo& assetInfo)
     {
         if (m_filterCallback && m_filterCallback(assetInfo))
         {
-            int index = findData(QVariant(QString(assetInfo.m_assetId.ToString<AZStd::string>().c_str())));
-            if (index >= 0 && index < count())
+            int index = findData(QVariant(QString(assetId.ToString<AZStd::string>().c_str())));
```

Comment:
Should we add a check to ensure assetId is valid?

Reply:
I would be more concerned about the index being invalid. Theoretically an invalid ID is a valid selection and something the user could handle upon receiving the signal.